backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -199,7 +199,9 @@ public String getNeededData() {
 
         if (needDatum != null) {
             return needDatum;
-        } else if (entries.size() == 1 && entries.elementAt(0) instanceof SyncEntry) {
+        } else if (entries.size() == 1
+                && entries.elementAt(0) instanceof SyncEntry
+                && ((SyncEntry)entries.elementAt(0)).getSyncPost().isRelevant(evalContext)) {
```

Comment:
Good call, added https://github.com/dimagi/commcare/pull/302/commits/28b607158cb72ae8ff480cebcfe35a568b732fe4 to error out if the `sync-request` suite xml entry doesn't have a `<post>` element